include/mbedtls/pk_info.h

```diff
@@ -0,0 +1,273 @@
+/**
+ * \file pk_info.h
+ *
+ * \brief Public Key cryptography abstraction layer: object interface
+ *
+ *  This file contains the info structure interface used by developers to
+ *  provide engine-specific implementations of opaque key handling functions.
```

Comment:
Suggestion: Afaik, `engine` is an established PKCS11-term, but I'd still like a more generic wording like `... used by developers to provide target-specific implementations of opaque key handling functions` more in this introduction.

Reply:
Done

Reply:
“Engine” is not a PKCS#11 term. The term is actually mostly popular from OpenSSL (but I'm pretty sure I've seen it elsewhere before). It's a convenient word and not overloaded (unlike “driver”).

Regardless, you're right about the wording: they aren't “engine-specific implementations”, the specific implementations are the engines.